test/ipns-pubsub.js

```diff
@@ -53,14 +52,12 @@ describe('ipns-pubsub', function () {
 
   // Connect nodes and wait for republish
   before(async function () {
-    await nodes[0].api.swarm.connect(nodes[1].api.peerId.addresses[0])
-
     // TODO: go-ipfs needs two nodes in the DHT to be able to publish a record
-    // Remove this when js-ipfs has a DHT
-    await nodes[0].api.swarm.connect(nodes[2].api.peerId.addresses[0])
-
-    console.log('wait for republish as we can receive the republish message first') // eslint-disable-line
-    await delay(60000)
+    // Remove the second connect when js-ipfs runs a DHT server
+    await Promise.all([
+      nodes[0].api.swarm.connect(nodes[1].api.peerId.addresses[0]),
+      nodes[0].api.swarm.connect(nodes[2].api.peerId.addresses[0])
+    ])
```

Comment:
In the IPNS over pubsub flow, there is a `initialBroadcastInterval` of 60 seconds. If we don't wait for it, the tests below might be impacted by the initial broadcast and not by the publish that happened in the test.
I think the solution is to allow this `initialBroadcastInterval` to be configured, both in go and js

Reply:
Actually, It can be configured in JS: https://github.com/ipfs/js-ipfs/blob/master/packages/ipfs/src/core/ipns/republisher.js#L72-L78

Reply:
Does the initial broadcast **only** happen after 60 seconds? The tests complete in < 5 seconds.

Reply:
> Does the initial broadcast only happen after 60 seconds?

By default yes. I remember it could happen to have false positives when this got in. But maybe, that's when we were also testing with the DHT

Reply:
Maybe this was due to the tests originally taking a very long time? Since the whole suite now finishes in under 5 seconds the first republish shouldn't be doing anything. I should probably remove the timeouts in these tests. If they do ever start taking 1 minute to resolve this could potentially create false positives.

Reply:
That sounds good!

Reply:
I removed the larger timeouts and added smaller ones for the startup and initial connects. With the updated timeouts the whole suite (aside from teardown) has a max allowed time of 45 seconds (30 for boot and 5*3 for tests).